src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -0,0 +1,450 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.Intrinsics
+{
+    /// <summary>Defines a single instruction, multiple data (SIMD) vector type.</summary>
+    /// <typeparam name="TSelf">The type that implements the interface.</typeparam>
+    /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+    internal interface ISimdVector<TSelf, T>
```

Comment:
I went with `ISimdVector` to avoid conflicting with the more general `IVector` that might be desirable for ML or Tensor like scenarios.

Reply:
This is also a place where "higher kinded types" would be beneficial.

It would allow us to specify that `TSelf` is itself generic and put constraints on that required generic parameter (e.g. `TSelf<T>`)